salt/states/file.py

```diff
@@ -4570,6 +4570,375 @@ def replace(name,
     return ret
 
 
+def keyvalue(
+        name,
+        key=None,
+        value=None,
+        key_values=None,
+        separator="=",
+        append_if_not_found=False,
+        prepend_if_not_found=False,
+        search_only=False,
+        show_changes=True,
+        ignore_if_missing=False,
+        count=1,
+        uncomment=None,
+        key_ignore_case=False,
+        value_ignore_case=False):
+    '''
+    Key/Value based editing of a file.
+
+    .. versionadded:: Neon
+
+    This function differs from ``file.replace`` in that it is able to search for
+    keys, followed by a customizable separator, and replace the value with the
+    given value. Should the value be the same as the one already in the file, no
+    changes will be made.
+
+    Either supply both ``key`` and ``value`` parameters, or supply a dictionary
+    with key / value pairs. It is an error to supply both.
+
+    name
+        Name of the file to search/replace in.
+
+    key
+        Key to search for when ensuring a value. Use in combination with a
+        ``value`` parameter.
+
+    value
+        Value to set for a given key. Use in combination with a ``key``
+        parameter.
+
+    key_values
+        Dictionary of key / value pairs to search for and ensure values for.
+        Used to specify multiple key / values at once.
+
+    separator : "="
+        Separator which separates key from value.
+
+    append_if_not_found : False
+        Append the key/value to the end of the file if not found. Note that this
+        takes precedence over ``prepend_if_not_found``.
+
+    prepend_if_not_found : False
+        Prepend the key/value to the beginning of the file if not found. Note
+        that ``append_if_not_found`` takes precedence.
+
+    search_only : False
+        Only search, do not actually replace. This is effectively a dry-run.
+        Functionally almost the same as running the state with test=True, which
```

Comment:
Does it make sense to have a separate flag then? What exactly distinguishes this from `test=True`?

Reply:
You are absolutely right @cachedout ! Admittedly, I originally wrote this function as an execution module where the need for a ``search_only`` parameter made sense just like it does in the ``file.replace`` execution module.

Back then it was a nice way of seeing which minions would end up with changes and which had already had their config files updated. Ultimately it seemed to make a lot more sense to put this into a state module. I suppose I documented it's behaviour and never realised it really should be taken out.

I'll go ahead and clean this up.

Reply:
Great! Thanks!